rapids-cmake/export/export.cmake

```diff
@@ -187,6 +225,34 @@ function(rapids_export type project_name)
     set(_RAPIDS_PROJECT_NAMESPACE ${_RAPIDS_NAMESPACE})
   endif()
 
+  if(_RAPIDS_COMPONENTS AND NOT _RAPIDS_COMPONENTS_EXPORT_SET)
+    message(FATAL_ERROR "rapids_export(${type} ${project_name} is missing COMPONENTS_EXPORT_SET as COMPONENTS was provided."
+    )
+  endif()
+  if(_RAPIDS_COMPONENTS_EXPORT_SET AND NOT _RAPIDS_COMPONENTS)
+    message(FATAL_ERROR "rapids_export(${type} ${project_name} is missing COMPONENTS as COMPONENTS_EXPORT_SET was provided."
+    )
+  endif()
+
+  if(_RAPIDS_COMPONENTS AND _RAPIDS_COMPONENTS_EXPORT_SET)
+    include("${rapids-cmake-dir}/export/detail/component.cmake")
+    set(_RAPIDS_HAS_COMPONENTS TRUE)
+    set(_RAPIDS_COMPONENTS_EXPORT_SET_NICE_NAMES)
+
+    foreach(comp comp_export_set IN ZIP_LISTS _RAPIDS_COMPONENTS _RAPIDS_COMPONENTS_EXPORT_SET)
+      string(REGEX REPLACE "(${project_name}[-_])|([-_]?${comp}[-_]?)|([-_]?export[s]?)" ""
```

Comment:
Should we document this transformation (depending on whether the nice names are intended to be public as I asked below)?

Reply:
I don't think we should since it could change and I don't want people to assume certain transformation